File: backend/agent/llm.py
"""LLM 客户端：混合模式的开关。

- 未配置 API Key：llm_enabled=False，调用方走规则/模板兜底，保证零依赖可跑。
- 配置 API Key：调用真实大模型做意图理解与文案润色。

对外暴露统一的 complete_json / complete_text 接口，调用方无需关心后端。
"""
from __future__ import annotations

import json

from ..config import settings


class LLMClient:
    def __init__(self) -> None:
        self.enabled = settings.llm_enabled

    async def complete_json(self, system: str, user: str) -> dict | None:
        """请模型返回 JSON。未启用或失败时返回 None，由调用方兜底。"""
        if not self.enabled:
            return None
        try:
            text = await self._call(system, user)
            # 容错：截取首个 { 到末个 } 之间的内容
            start, end = text.find("{"), text.rfind("}")
            if start == -1 or end == -1:
                return None
            return json.loads(text[start:end + 1])
        except Exception:
            return None

    async def complete_text(self, system: str, user: str) -> str | None:
        """请模型返回自由文本（用于文案润色）。失败返回 None。"""
        if not self.enabled:
            return None
        try:
            return await self._call(system, user)
        except Exception:
            return None

    async def _call(self, system: str, user: str) -> str:
        """调用 OpenAI 兼容格式 API（DeepSeek / OpenAI 均可）。"""
        import httpx

        headers = {
            "Authorization": f"Bearer {settings.LLM_API_KEY}",
            "Content-Type": "application/json",
        }
        payload = {
            "model": settings.LLM_MODEL,
            "max_tokens": 1024,
            "messages": [
                {"role": "system", "content": system},
                {"role": "user", "content": user},
            ],
        }
        async with httpx.AsyncClient(timeout=30) as client:
            resp = await client.post(
                f"{settings.LLM_BASE_URL}/chat/completions",
                headers=headers, json=payload,
            )
            resp.raise_for_status()
            data = resp.json()
            return data["choices"][0]["message"]["content"]


llm = LLMClient()

File: functions/api/llm.js
import { json, missing } from "../_lib.js";

export async function onRequestPost({ request, env }) {
  if (!env.LLM_API_KEY) return missing("LLM_API_KEY");

  const body = await request.json().catch(() => ({}));
  const base = (env.LLM_BASE_URL || "https://api.deepseek.com/v1").replace(/\/+$/, "");
  const requestedTokens = Number(body.max_tokens || (body.jsonMode ? 768 : 320));
  const maxTokens = Math.max(64, Math.min(requestedTokens, body.jsonMode ? 900 : 500));
  const systemPrompt = body.jsonMode
    ? `${body.systemPrompt || ""}\n\n只输出合法 JSON。不要 markdown，不要解释，不要推理过程。`
    : `${body.systemPrompt || ""}\n\n直接给最终答复，不要展开推理过程。`;
  const payload = {
    model: env.LLM_MODEL || "deepseek-v4-flash",
    max_tokens: maxTokens,
    messages: [
      { role: "system", content: systemPrompt },
      { role: "user", content: body.userContent || "" },
    ],
  };
  if (body.jsonMode) payload.response_format = { type: "json_object" };

  const upstream = await fetch(`${base}/chat/completions`, {
    method: "POST",
    headers: {
      "authorization": `Bearer ${env.LLM_API_KEY}`,
      "content-type": "application/json",
    },
    body: JSON.stringify(payload),
  });
  const data = await upstream.json().catch(() => ({}));
  if (!upstream.ok) return json({ error: data.error?.message || upstream.statusText }, upstream.status);

  return json({ content: data.choices?.[0]?.message?.content || "" });
}
